independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java

```diff
@@ -61,11 +61,30 @@ protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
     }
 
     static boolean addDependencyInModel(Model model, AppArtifactCoords coords, boolean managed) {
-        StringBuilder newDependency = new StringBuilder()
-                .append("    implementation '")
-                .append(coords.getGroupId())
-                .append(":")
-                .append(coords.getArtifactId());
+        boolean isBOM = "pom".equals(coords.getType());
+        StringBuilder newDependency;
+        if (isBOM) {
+            // Check if BOM is not included already
+            String resolvedPlatform = String
+                    .format("%s:%s:%s", getProperty(model, "quarkusPlatformGroupId"),
+                            getProperty(model, "quarkusPlatformArtifactId"),
+                            getProperty(model, "quarkusPlatformVersion"));
+            String thisBOM = String.format("%s:%s:%s", coords.getGroupId(), coords.getArtifactId(), coords.getVersion());
```

Comment:
Do we really want to compare with the version?

Reply:
The version will (or should) always be the same here

Reply:
@gastaldi as @gsmet pointed out, I think we should have the same bahavior as for extensions which is to skip it if it's already there whatever the version.

Reply:
@ia3andy although working with the model as a big String in the case of Gradle is a PITA, I think that might make sense, let's do that in a separate PR